mapie/quantile_regression.py

```diff
@@ -0,0 +1,414 @@
+            sample_weight,
+            X_train,
+            y_train
+        )
+        y_train = cast(NDArray, y_train)
+
+        # Initialization
+        self.estimators_ = {}
+
+        # Work
+        y_calib_preds = np.full(
+            shape=(3, len(y_calib)),
+            fill_value=np.nan
+        )
+
+        name_estimator = estimator.__class__.__name__
+        for i, alpha_ in enumerate(alpha):
+            estimator_cloned = clone(estimator)
+            params = {
+                self.quantile_estimator_params[
+                    name_estimator
+                    ]["alpha_name"]: alpha_
+            }
+            estimator_cloned.set_params(**params)
+            estimator_cloned_ = fit_estimator(
+                estimator_cloned, X_train, y_train, sample_weight
+            )
+            self.estimators_[alpha_] = estimator_cloned_
+            y_calib_pred = estimator_cloned_.predict(X_calib)
+            check_nan_in_aposteriori_prediction(y_calib_pred)
+            y_calib_preds[i] = y_calib_pred
+
+        self.conformity_scores_ = np.full(
+                shape=(3, len(y_calib)),
+                fill_value=np.nan
+            )
+        self.conformity_scores_[0] = y_calib_preds[0]-y_calib
+        self.conformity_scores_[1] = y_calib-y_calib_preds[1]
+        self.conformity_scores_[2] = np.max(
+                [
+                    self.conformity_scores_[0],
+                    self.conformity_scores_[1]
+                ], axis=0
+            )
+        return self
+
+    def predict(
+        self,
+        X: ArrayLike,
+        symmetry: Optional[bool] = True,
+    ) -> Union[NDArray, Tuple[NDArray, NDArray]]:
+        """
+        Predict target on new samples with confidence intervals.
+        Residuals from the training set and predictions from the model clones
+        are central to the computation.
+        Prediction Intervals for a given ``alpha`` are deduced from the
+        quantile regression at the alpha values: alpha/2, 1 - (alpha/2)
+        while adding a constant based uppon their residuals.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Test data.
+        symmetry : Optional[bool], optional
+            Deciding factor to whether to find the quantile value for
+            each residuals separatly or to use the maximum of the two
+            combined. This results in having either symmetric constants
+            added for both upper and lower bounds or not.
+
+        Returns
+        -------
+        Union[NDArray, Tuple[NDArray, NDArray]]
+
+        - NDArray of shape (n_samples,) if alpha is None.
+
+        - Tuple[NDArray, NDArray] of shapes
+        (n_samples,) and (n_samples, 2, n_alpha) if alpha is not None.
+
+            - [:, 0, :]: Lower bound of the prediction interval.
+            - [:, 1, :]: Upper bound of the prediction interval.
+        """
+        n = self.n_samples_calib
```

Comment:
Move this line after the checks to be as consistent with `MapieRegressor` as possible.